thoth/lab/inspection.py

```diff
@@ -457,13 +457,210 @@ def show_categories(inspection_df: pd.DataFrame):
+    )
+
+    py.iplot(fig)
+    # Scatter build duration
+    fig = inspection.create_duration_scatter(
+        df_duration_batches[batch_identifier], "build_duration", title="InspectionRun build duration"
+    )
+
+    py.iplot(fig)
+    # Histogram
+    fig = inspection.create_duration_histogram(df_duration_batches[batch_identifier], ["job_duration"])
+
+    py.iplot(fig)
+
+
+def create_scatter_plots_for_multiple_batches(
+    data: pd.DataFrame,
+    columns: Union[str, List[str]] = None,
+    list_batches: List[str] = [],
+    title_scatter: str = "Scatter plot",
+    x_label: str = " ",
+    y_label: str = " ",
+):
+    """Create Scatter plots for multiple batches."""
+    columns = columns if columns is not None else data[columns].columns
+
+    figure = {
+        "data": [
+            {"x": data[columns[0] + "_" + batch], "y": data[columns[1] + "_" + batch], "name": batch, "mode": "markers"}
+            for batch in list_batches
+        ],
+        "layout": {"title": title_scatter, "xaxis": {"title": x_label}, "yaxis": {"title": y_label}},
+    }
+
+    return figure
+
+
+def evaluate_statistics(df_batch: pd.DataFrame, parameter_name: str) -> Dict:
+    """Evaluate statistical quantities for plots."""
+    cv = df_batch[parameter_name].std() / df_batch[parameter_name].mean() * 100
+    std_error = df_batch[parameter_name].std() / np.sqrt(df_batch[parameter_name].shape[0])
+    std = df_batch[parameter_name].std()
+    median = df_batch[parameter_name].median()
+    q = df_batch[parameter_name].quantile([0.25, 0.75])
+    q1 = q[0.25]
+    q3 = q[0.75]
+    IQR = q3 - q1
+    maxr = df_batch[parameter_name].max()
+    minr = df_batch[parameter_name].min()
+
+    return {
+        "cv": cv,
+        "std_error": std_error,
+        "std": std,
+        "median": median,
+        "q1": q1,
+        "q3": q3,
+        "iqr": IQR,
+        "max": maxr,
+        "min": minr,
+    }
+
+
+def evaluate_statistics_per_parameter_per_batch(dict_of_df_batches: dict, parameter: str) -> dict:
+    """Aggregate results of statistics per parameter from different inspection batches."""
+    results_statistics = {}
+    for identifier in identifier_inspection:
+        results_statistics[identifier] = evaluate_statistics(
+            df_batch=dict_of_df_batches[identifier], parameter_name=parameter
+        )
+
+    results = {}
+
+    for quantity in results_statistics[identifier].keys():
+        results[quantity] = [values[quantity] for values in results_statistics.values()]
+
+    return results
+
+
+def plot_batches_statistics_interpolated_single_parameter(
+    parameter_selected: str, colour_list: List, quantities: List, title_ylabel: str = " "
+):
+    """Plot interpolated statistics for different batches for a specific parameter."""
+    if len(colour_list) != len(quantities):
+        logger.exception(f"List of statistical quantities and List of colours need to have the same length!")
```

Comment:
I don't see this call being used in any try-except block (no exception is captured, no exception will be reported):

```python
In [1]: import logging

In [2]: logging.getLogger(__name__).exception("foo")
foo
NoneType: None

In [3]: 

```

Here and bellow.